### PR TITLE
Move dev dependencies to requirements-dev.txt (#61)

### PR DESCRIPTION
**Pull request by mehaase**
I used pip-tools to generate a requirements file. Also added a
Makefile to emulate how the requests package is laid out. Added
some documentation for using pip-tools. Finally, removed the
attrs dependency which wasn't really being used heavily enough
to justify its inclusion on this project.

### CODE REVIEW

**Comment by mehaase** [comment]
This isn't working as intended, yet. It is supposed to install trio-websocket in the development environment, so I put `-e .` in the dependencies file, but `pip-compile` expands this to a full path that only exists on my machine `-e file:///Volumes/Case_Sensitive/trio-websocket`. There's [an issue filed on `pip-tools`](https://github.com/jazzband/pip-tools/issues/204) but it looks like it's a complex interaction between `pip` and `pip-tools` that isn't likely to be solved anytime soon. I'll take another shot at this tomorrow.

**Comment by belm0** [comment]
It's certainly possible to get by without using "`-e .`".  Just prefix commands with appropriate setting of PYTHONPATH.
```makefile
test:
	PYTHONPATH=. pytest --cov=trio_websocket
```

File: requirements-dev.in [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# This file specifies development dependencies. It is processed by pip-compile
+# to generate requirements-dev.txt. See the developer docs for more information.
+-e .
```
Comment (belm0) [inline requirements-dev.in]: As mentioned, you can remove this and rely on dev command invocations to set PYTHONPATH.  Easy now that you have a makefile.

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#
+# This file is autogenerated by pip-compile
+# To update, run:
+#
+#    pip-compile --output-file requirements-dev.txt requirements-dev.txt
```
[the remark this answers is not in the document]
Comment (mehaase) [reply requirements-dev.txt]: Yes, good catch! Weird because this is a generated file. But I double checked my latest commit and this is fixed.

**Comment by mehaase** [comment]
This still doesn't work for RTD builds. RTD does not use the Makefile, so it does not know where `trio-websocket` is. RTD can be configured to install from `setup.py`, but then it installs packages that conflict with `requirements-dev.txt`. I've pushed [a quick fix on master](https://github.com/HyperionGray/trio-websocket/commit/ce4adf3e958c166fc5c5b7ec1e81693478afb531) that abandons pip tools and uses a hand-written `requirements-dev.txt` instead.

**Pull request merged**